core-web/libs/sdk/nextjs/src/lib/components/DotcmsLayout/DotcmsLayout.tsx

```diff
@@ -50,23 +52,25 @@ export function DotcmsLayout(props: DotcmsPageProps) {
     const router = useRouter();
     const pathname = usePathname();
 
-    const { rowsRef, isInsideEditor } = usePageEditor({
-        reloadFunction: router.refresh,
-        pathname
+    const client = sdkDotPageEditor.createClient({
+        onReload: router.refresh
     });
 
-    const addRowRef = (el: HTMLDivElement) => {
-        if (el && !rowsRef.current.includes(el)) {
-            rowsRef.current.push(el);
-        }
-    };
+    useEffect(() => {
+        client.init();
+        client.updateNavigation(pathname);
 
-    entity.isInsideEditor = isInsideEditor;
+        return () => {
+            client.destroy();
+        };
+    }, [client, pathname]);
+
+    entity.isInsideEditor = client.isInsideEditor;
```

Comment:
This should be the other way around... we init the whole thing if this is inside editor.

Did you solve the issues with the multiples re-renders? If not... that `useEffect` looks very suspicious to me.

We might need to hop in a call and talk about the whole client API.